Extensions/dist/extensions.js

```diff
@@ -2,158 +2,213 @@
 
 var fs = require('fs');
 
-const extensionPath = process.argv[2] || '..';
-const distributionPath = process.argv[3] || '.';
-
-processExtensions();
-
-function processExtensions() {
-  var extensionFiles = fs.readdirSync(extensionPath);
-  var extensions = [];
-
-  extensionFiles.forEach(function(fileName) {
-    // Look only for files for the form [a-z_]+.js, which exclues icon and css
-    // files
-    var fileNameParts = fileName.split('.');
-    if (fileNameParts.length !== 2) {
-      return;
-    }
-    var fileType = fileNameParts[1];
-    if (fileType !== 'js') {
-      return;
-    }
-    var extensionId = fileNameParts[0];
-
-    extensions.push(processExtension(extensionId));
-  });
-
-  writeGalleryFile(extensions);
-  writeListFile(extensions);
+/**
+ * Build the extensions in `extensionPath`, writing the distribution json files
+ * into `distributionPath`.
+ * @param {String} extensionPath
+ * @param {String} distributionPath
+ * @exported
+ */
+function build(extensionPath, distributionPath) {
+	var builder = new ExtensionBuilder(extensionPath, distributionPath);
+	builder.processExtensions();
 }
-
-function readExtensionFile(path) {
-  return fs.readFileSync(extensionPath + '/' + path, {
-    encoding: 'utf8'
-  });
-}
-
-function writeDistributionFile(path, contents) {
-  fs.writeFileSync(distributionPath + '/' + path, contents);
-}
-
-function writePageFile(path, content) {
-  writeDistributionFile('page/' + path, content);
-}
-
-function existsExtensionFile(path) {
-  return fs.existsSync(extensionPath + '/' + path);
+exports.build = build;
+
+/**
+ * @constructor
+ * @param {String} extensionPath - Path from which to read extension files
+ * @param {String} distributionPath - Path in whichw to write extension distribution files
+ */
+function ExtensionBuilder(extensionPath, distributionPath) {
+	this.extensionPath = extensionPath;
+	this.distributionPath = distributionPath;
 }
 
+/**
+ * Process all extensions found in `extensionPath`
+ */
+ExtensionBuilder.prototype.processExtensions = function() {
+	var extensionFiles = fs.readdirSync(this.extensionPath);
+	var extensions = [];
+
+	extensionFiles.forEach(function(fileName) {
+		// Look only for files for the form [a-z_]+.js, which exclues icon and css
+		// files
+		var fileNameParts = fileName.split('.');
+		if (fileNameParts.length !== 2) {
+			return;
+		}
+		var fileType = fileNameParts[1];
+		if (fileType !== 'js') {
+			return;
+		}
+		var extensionId = fileNameParts[0];
+
+		extensions.push(this.processExtension(extensionId));
+	}.bind(this));
+
+	this.writeGalleryFile(extensions);
+	this.writeListFile(extensions);
+};
+
+/**
+ * Read a file in the extensions directory
+ * @param {String} path - Relative path of file in directory
+ * @return {String} content in file
+ */
+ExtensionBuilder.prototype.readExtensionFile = function(path) {
+	return fs.readFileSync(this.extensionPath + '/' + path, {
```

Comment:
Thoughts on string concat vs [`path.join()`](https://nodejs.org/api/path.html#path_path_join_path1_path2).  As far as I know it shouldn't make a difference in this case (like, we don't have to worry about Unix vs Win path separators, Node internals will handle that for us), except to future-proof these functions a bit.